Server/src/services/getDataService.ts

```diff
@@ -0,0 +1,122 @@
+import { EntityManager } from "typeorm";
+
+const obtainDataModel = require('../models/SelectQueryDatabase');
+
+interface IDataRequestModel {
+    datasetId: number
+    material: string
+    firstName: string
+    lastName: string
+    year: number
+    categoryId: number
+    subcategoryId: number
+}
+
+export const retrieveData = async (req) => {
+
+    const request: IDataRequestModel = req.query
+    let datasetReceived = request.datasetId;
+    let materialReceived = request.material;
+    let firstNameReceived = request.firstName;
+    let lastNameReceived = request.lastName;
+    let yearReceived = request.year;
+    let categoryReceived = request.categoryId;
+    let subcategoryReceived = request.subcategoryId;
+    let setOfData = [];
+
+
+    // To avoid checks for both first and last name, make a bool for author being entered
+    let authorEntered = false
+    if (firstNameReceived != undefined && lastNameReceived != undefined) {
+        authorEntered = true
+    }
+
+    //As subcategory requires an accompanying category, make a bool for
+    //Both category and subcategory being entered
+    let subcategoryEntered = false
+    if (categoryReceived != undefined && subcategoryReceived != undefined) {
+        subcategoryEntered = true
+    }
+    //Subcategory queries must always be listed above category queries, 
+    //as a subcategory query contains both a category and subcategory entry
+    setOfData = await obtainDataModel.getDataFromAuthorSubcategory(firstNameReceived, lastNameReceived, categoryReceived, subcategoryReceived);
+    if (datasetReceived != undefined) {
+        setOfData = await obtainDataModel.getDataFromDataset(datasetReceived);
+    }
+    else if (materialReceived != undefined) {
```

Comment:
Lets avoid these if statements. Switch statements are better.... Even better - https://refactoring.guru/replace-conditional-with-polymorphism. Follows the strategy pattern with polymorphism. I can assist with this if its required.